### PR TITLE
Feat Query-Scheduler

### PR DESCRIPTION
**Pull request by marquezika**
**What this PR does**:

This MR adds the component Query-Scheduler - this makes scaling query frontend easier.

_Reference: [scaling-query-frontend](https://cortexmetrics.io/docs/operations/scaling-query-frontend/#query-scheduler)._

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by ShuzZzle — CHANGES_REQUESTED** [review]
Hi,

Thank you for your contribution. Could you please do the following?

- [ ] Rebase against current master
- [ ] implement suggestions
- [ ] add ci test for query-scheduler to `test-deployment-values.yaml` or both

File: templates/query-scheduler/query-scheduler-servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{{- if and .Values.query_scheduler.enabled .Values.query_scheduler.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "cortex.querySchedulerFullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "cortex.querySchedulerLabels" . | nindent 4 }}
+    {{- if .Values.query_scheduler.serviceMonitor.additionalLabels }}
+{{ toYaml .Values.query_scheduler.serviceMonitor.additionalLabels | indent 4 }}
```
Comment (ShuzZzle) [inline templates/query-scheduler/query-scheduler-servicemonitor.yaml:10]: i know the other service monitors are exactly like this but I think we should move everything to nindent. All of the other stuff already is

File: templates/query-scheduler/query-scheduler-servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    {{- end }}
+  {{- if .Values.query_scheduler.serviceMonitor.annotations }}
+  annotations:
+{{ toYaml .Values.query_scheduler.serviceMonitor.annotations | indent 4 }}
```
Comment (ShuzZzle) [inline templates/query-scheduler/query-scheduler-servicemonitor.yaml:14]:
```suggestion
    {{- toYaml .Values.query_scheduler.serviceMonitor.annotations | nindent 4 }}
```

**Comment by marquezika** [comment]
Hi @ShuzZzle.

Sure, I gonna implement these changes - thanks for reviewing.

**Comment by ShuzZzle** [comment]
@marquezika Need some help? :)

**Comment by nschad** [comment]
Hey @marquezika,

Do you still wanna do this? I already rebased the branch on the current master otherwise I'm happy to finish this PR

**Review by nschad — APPROVED** [review]

**Review by nschad — CHANGES_REQUESTED** [review]
/

**Review by nschad — APPROVED** [review]
@kd7lxl I think this is okay like this

might need some tests tho

**Review by kd7lxl — COMMENTED** [review]
LGTM. Yes, please add some CI values to exercise it in the tests. Then we can merge.

**Pull request merged**